web/src/compose_reply.ts

```diff
@@ -86,7 +86,10 @@ export function respond_to_message(opts: {
             const first_operator = first_term.operator;
             const first_operand = first_term.operand;
 
-            if (first_operator === "stream" && !stream_data.is_subscribed_by_name(first_operand)) {
+            if (
+                first_operator === "stream" &&
+                !stream_data.is_subscribed(Number.parseInt(first_operand, 10))
+            ) {
                 compose_actions.start({
```

Comment:
Should this logic just be using `narrow_state.stream_id`? That might be a prep refactor to reduce the madness here.

(I have a vague recollection of us discussing this weird block of code previously).

Reply:
Ah yeah that would work very well, thanks!